cmd/agent/main_windows.go

```diff
@@ -37,6 +52,28 @@ func main() {
 		}
 	}
 	defer log.Flush()
+	setConsoleCtrlHandler.Call(
+		syscall.NewCallback(func(controlType uint) uint {
+			var sigStr string
+			switch controlType {
+			case ctrlCEvent:
+				sigStr = "CTRL+C"
+			case ctrlBreakEvent:
+				sigStr = "CTRL+BREAK"
+			case ctrlCloseEvent:
+				sigStr = "CTRL+CLOSE"
+			case ctrlLogOffEvent:
+				sigStr = "CTRL+LOG_OFF"
+			case ctrlShutdownEvent:
+				sigStr = "CTRL+SHUTDOWN"
+			}
+			log.Infof("Received control event '%s', shutting down...", sigStr)
+			// signals.Stopper <- true
+			// Completely bypass the run command stop logic as it takes too long to stop
+			app.StopAgent()
```

Comment:
Do we need to take of this in other agents, or will the core agent stop the Windows services in this `StopAgent` function?

Reply:
To clarify, here is the layout.
In K8S:
- All agent binaries are run directly
In Docker/ECS:
- Core agent binary is run directly
- trace-agent/process-agent are run through windows services.